data_processing/wdl/scatterGuppy.wdl

```diff
@@ -0,0 +1,348 @@
+		rm ~{file_to_split_tar}
+
+		mkdir input
+
+		# filter out only fast5 files
+		for FILE in `find tmp/ -name "*.fast5"`
+		do
+			mv $FILE input
+		done
+
+
+		# move files into folder until exceeds desired_size_GB
+		# then tar contents of folder
+		OUTPUT_IDX=0
+		OUTPUT_DIR=fast5_tar_$OUTPUT_IDX
+		mkdir $OUTPUT_DIR
+		for FILE in `ls input`
+		do
+			size=$(du -s -BG $OUTPUT_DIR | sed 's/G.*//')
+			echo $size
+			if (( $size > ~{desired_size_GB} ))
+			then
+				tar -cvf fast5_tarball_$OUTPUT_IDX.tar $OUTPUT_DIR/*
+				rm -r $OUTPUT_DIR
+				OUTPUT_IDX=$(($OUTPUT_IDX + 1))
+				OUTPUT_DIR=fast5_tar_$OUTPUT_IDX
+				mkdir $OUTPUT_DIR
+			fi
+			echo $(du -s -BG $OUTPUT_DIR | sed 's/G.*//')
+			mv input/$FILE $OUTPUT_DIR
+		done
+		tar -cvf fast5_tarball_$OUTPUT_IDX.tar $OUTPUT_DIR/*
+		rm -r $OUTPUT_DIR
+
+	>>>
+
+	output {
+		Array[File] split_fast5_tar = glob("*tar")
+	}
+
+	runtime {
+		memory: memSizeGB + " GB"
+		cpu: threadCount
+		disks: "local-disk " + diskSizeGB + " SSD"
+		docker: dockerImage
+		preemptible : preempts
+	}
+}
+
+task guppyGPU {
+	
+	input {
+	
+		File fast5_tar_file
+
+		String CONFIG_FILE = "dna_r9.4.1_450bps_modbases_5mc_cg_sup_prom.cfg"
+		Int READ_BATCH_SIZE = 250000
+		Int q = 250000
+
+		String dockerImage = "jiminpark/guppy-wdl:latest" 
+
+		String? additionalArgs
+
+		Int preempts = 3
+		Int memSizeGB = 64
+		Int threadCount = 10
+		Int extraDisk = 5
+		Int gpuCount = 1
+		Int maxRetries = 0
+		String gpuType = "nvidia-tesla-v100"
+		String nvidiaDriverVersion = "418.87.00"
+		String zones = "us-west1-b"
+	}
+
+	# calculate needed disk size
+	Int file_size = ceil(size(fast5_tar_file, "GB"))
+	Int diskSizeGB = 3 * file_size + extraDisk
+
+
+	command <<<
+		# Set the exit code of a pipeline to that of the rightmost command
+		# to exit with a non-zero status, or zero if all commands of the pipeline exit
+		set -o pipefail
+		# cause a bash script to exit immediately when a command fails
+		set -e
+		# cause the bash shell to treat unset variables as an error and exit immediately
+		set -u
+		# echo each line of the script to stdout so we can see what is happening
+		# to turn off echo do 'set +o xtrace'
+		set -o xtrace
+
+		## Extract tar file to 
+		mkdir input
+		
+		# place all extracted files into directory input
+		tar xvf "~{fast5_tar_file}" --directory input
+
+		mkdir output
+
+		# check if length of "additionalArgs" is zero
+
+		if [[ "~{additionalArgs}" == "" ]]
+		then
+			ADDITIONAL_ARGS=""
+		else
+			ADDITIONAL_ARGS="~{additionalArgs}"
+		fi
+
+
+		guppy_basecaller \
+			-i input/ \
+			-s output/ \
+			-c /opt/ont/guppy/data/"~{CONFIG_FILE}" \
+			--bam_out \
+			-x cuda:all:100% \
+			-r \
+			--read_batch_size "~{READ_BATCH_SIZE}" \
+			-q "~{q}" \
+			${ADDITIONAL_ARGS}
+
+	>>>
+
+
+	output {
+		File pass_bam = glob("output/pass/*.bam")[0]
+		File pass_fastq = glob("output/pass/*.fastq")[0]
+		File summary = glob("output/sequencing_summary.txt")[0]
```

Comment:
Did you ever check that Guppy only outputs one bam/fastq? If not, the comparison of your run against the prior runs is especially important.